File: leetcode/easy/3870-count-commas-in-range/solution.cpp
class Solution {
public:
    int countCommas(int n) {
        if (n < 1000) return 0;
        long long ans = 0;

        long long start = 1000;
        int commas = 1;

        while (start <= n){
            long long end = min((long long)n, start*1000-1);
            ans += (end - start + 1)*commas;

            start *= 1000;
            commas++;
        }

        return ans;

    }
};

File: leetcode/easy/3870-count-commas-in-range/solution.java
class Solution {
    public int countCommas(int n) {
        int count = 0;

        long start = 1000;
        long commas = 1;

        while (start <= n){
            long end = Math.min((long)n, start*1000-1);
            count += end - start + 1;

            start *= 1000;
            commas++;
        }

        return count;
    }
}